internal/server/providers/azure_test.go

```diff
@@ -0,0 +1,328 @@
+package providers
+
+import (
+	"testing"
+	"time"
+
+	"gopkg.in/square/go-jose.v2/jwt"
+	"gotest.tools/v3/assert"
+
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+const azureGroupResponse = `{
+	"@odata.context": "https://graph.microsoft.com/v1.0/$metadata#directoryObjects",
+	"value": [
+		{
+			"@odata.type": "#microsoft.graph.directoryRole",
+			"id": "aaa",
+			"deletedDateTime": null,
+			"description": "Can manage all aspects of Azure AD and Microsoft services that use Azure AD identities.",
+			"displayName": "Global Administrator",
+			"roleTemplateId": "bbb"
+		},
+		{
+			"@odata.type": "#microsoft.graph.group",
+			"id": "ccc",
+			"deletedDateTime": null,
+			"classification": null,
+			"createdDateTime": "2022-06-07T20:40:20Z",
+			"creationOptions": [],
+			"description": null,
+			"displayName": "Everyone",
+			"expirationDateTime": null,
+			"groupTypes": [],
+			"isAssignableToRole": null,
+			"mail": null,
+			"mailEnabled": false,
+			"mailNickname": "eee",
+			"membershipRule": null,
+			"membershipRuleProcessingState": null,
+			"onPremisesDomainName": null,
+			"onPremisesLastSyncDateTime": null,
+			"onPremisesNetBiosName": null,
+			"onPremisesSamAccountName": null,
+			"onPremisesSecurityIdentifier": null,
+			"onPremisesSyncEnabled": null,
+			"preferredDataLocation": null,
+			"preferredLanguage": null,
+			"proxyAddresses": [],
+			"renewedDateTime": "2022-06-07T20:40:20Z",
+			"resourceBehaviorOptions": [],
+			"resourceProvisioningOptions": [],
+			"securityEnabled": true,
+			"securityIdentifier": "qqq",
+			"theme": null,
+			"visibility": null,
+			"onPremisesProvisioningErrors": []
+		},
+		{
+			"@odata.type": "#microsoft.graph.group",
+			"id": "ccc",
+			"deletedDateTime": null,
+			"classification": null,
+			"createdDateTime": "2022-06-07T20:40:20Z",
+			"creationOptions": [],
+			"description": null,
+			"displayName": "Developers",
+			"expirationDateTime": null,
+			"groupTypes": [],
+			"isAssignableToRole": null,
+			"mail": null,
+			"mailEnabled": false,
+			"mailNickname": "eee",
+			"membershipRule": null,
+			"membershipRuleProcessingState": null,
+			"onPremisesDomainName": null,
+			"onPremisesLastSyncDateTime": null,
+			"onPremisesNetBiosName": null,
+			"onPremisesSamAccountName": null,
+			"onPremisesSecurityIdentifier": null,
+			"onPremisesSyncEnabled": null,
+			"preferredDataLocation": null,
+			"preferredLanguage": null,
+			"proxyAddresses": [],
+			"renewedDateTime": "2022-06-07T20:40:20Z",
+			"resourceBehaviorOptions": [],
+			"resourceProvisioningOptions": [],
+			"securityEnabled": true,
+			"securityIdentifier": "qqq",
+			"theme": null,
+			"visibility": null,
+			"onPremisesProvisioningErrors": []
+		}
+	]
+}`
+
+func TestSyncAzureProviderUser(t *testing.T) {
```

Comment:
```suggestion
func TestAzure_SyncProviderUser(t *testing.T) {
```

This is `Test[StructName]_[MethodName]` convention for naming tests.